app/(api)/_datalib/auth/verifyLogin.ts

```diff
@@ -0,0 +1,40 @@
+import bcrypt from 'bcrypt';
+import jwt, { Secret } from 'jsonwebtoken';
+
+/**
+ * Compares a provided password with a stored password hash.
+ * @param providedPassword The password provided by the user.
+ * @param storedHash The hash stored in the database.
+ * @returns True if the password matches the hash, false otherwise.
+ */
+export async function comparePasswordHash(
+  providedPassword: string,
+  storedHash: string
+): Promise<boolean> {
+  return bcrypt.compare(providedPassword, storedHash);
+}
+
+export async function verifyLogin(email: string, password: string) {
+  try {
+    // If unsuccessful login of user or password, return AuthenticationError.
+    // getUserByEmail TO BE IMPLEMENTED.
+    const user = await getUserByEmail(email);
+    if (!user) {
+      return { ok: false, error: 'AuthenticationError' };
```

Comment:
Make sure to be returning in the format:

```js
{
  ok: boolean,
  body: ... or null,
  error: ... or null
}
```

it keeps our returns a little more consistent

Reply:
Also, if you don't want to write the return statement a bunch, I recommend just throwing a custom Authentication Error. I define a couple custom errors in the `/(api)/_utils/response` folder.

This allows the try catch to handle it assuming you have the catch part display the error msg